packages/dds/tree/src/test/edit-manager/editManager.spec.ts

```diff
@@ -636,14 +678,24 @@ function runScenario(scenario: readonly ScenarioStep[]): void {
                 });
             } else { // step.type === "Receive"
                 const commit = trunk[client.ref];
-                client.manager.addSequencedChange(commit);
+                const delta = client.manager.addSequencedChange(commit);
                 // If the change came from this client
                 if (commit.sessionId === step.client.toString()) {
+                    assert.deepEqual(delta, Delta.empty);
                     // Discard the local change
                     client.localChanges.shift();
                     // Do not update the intentions
                 } else {
-                    // Update the intentions known to this client
+                    const localIntentions = ([] as number[]).concat(
+                        ...client.localChanges.map((c) => c.change.intentions),
+                    );
```

Comment:
Looks like in ES2019 there is a `flatMap` method on Array. Would be nice if we put something like that in a util file.